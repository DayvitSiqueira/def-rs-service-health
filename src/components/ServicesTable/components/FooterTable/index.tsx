import PagesSelect from '@/components/PagesSelect'
import Pagination from '@/components/Pagination'

export default function FooterTable() {
  return (
    <div className="flex flex-col items-end justify-end w-full mt-6 gap-6">
      <div className="flex items-end justify-between w-full mt-6 gap-6">
        <div className="flex items-center gap-2">
          <p className="text-sm text-gray-500 dark:text-gray-400 whitespace-nowrap">
            Resultados por p√°gina:
          </p>
          <PagesSelect />
        </div>

        <Pagination />
      </div>
    </div>
  )
}
