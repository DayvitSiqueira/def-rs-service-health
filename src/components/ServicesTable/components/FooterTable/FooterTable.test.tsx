import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import FooterTable from './index' // Update the import path accordingly

describe('FooterTable', () => {
  const mockOnChange = jest.fn()

  it('renders the FooterTable component correctly', () => {
    render(<FooterTable onChange={mockOnChange} />)
    const textElement = screen.getByText('Resultados por pÃ¡gina:')

    expect(textElement).toBeInTheDocument()
  })

  it('renders the PagesSelect component within FooterTable', () => {
    render(<FooterTable onChange={mockOnChange} />)
    const selectElement = screen.getByRole('combobox')

    expect(selectElement).toBeInTheDocument()
  })

  it('handles the onChange event of PagesSelect', () => {
    render(<FooterTable onChange={mockOnChange} />)
    const selectElement = screen.getByRole('combobox')

    fireEvent.change(selectElement, { target: { value: '15' } })

    expect(mockOnChange).toHaveBeenCalledTimes(1)
    expect(mockOnChange).toHaveBeenCalledWith('15')
  })

  it('matches the snapshot', () => {
    const { container } = render(<FooterTable />)
    expect(container).toMatchSnapshot()
  })
})
