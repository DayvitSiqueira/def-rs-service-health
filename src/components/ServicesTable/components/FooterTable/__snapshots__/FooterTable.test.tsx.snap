// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterTable Component matches the snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-end justify-end w-full mt-6 gap-6"
  >
    <div
      class="flex items-end justify-between w-full mt-6 gap-6"
    >
      <div
        class="flex items-center gap-2"
      >
        <p
          class="text-sm text-gray-500 dark:text-gray-400 whitespace-nowrap"
        >
          Resultados por página:
        </p>
        <select
          class="max-w-lg bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white"
        >
          <option
            value="10"
          >
            10
          </option>
          <option
            value="15"
          >
            15
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;

exports[`FooterTable matches the snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-end justify-end w-full mt-6 gap-6"
  >
    <div
      class="flex items-end justify-between w-full mt-6 gap-6"
    >
      <div
        class="flex items-center gap-2"
      >
        <p
          class="text-sm text-gray-500 dark:text-gray-400 whitespace-nowrap"
        >
          Resultados por página:
        </p>
        <select
          class="max-w-lg bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white"
        >
          <option
            value="10"
          >
            10
          </option>
          <option
            value="15"
          >
            15
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
